### PR TITLE
Decompile `no3` `EntityExplosion3`

### PR DESCRIPTION
**Pull request by JoshSchreuder**
https://decomp.me/scratch/XWQUk

I'm sure there's some non-idiomatic code I've written here so looking forward to rewriting some 😄

### CODE REVIEW

File: src/st/no3/4AFF0.c [hunk 1]
```diff
@@ -319,7 +319,127 @@ void EntityUnkId31(Entity* self) {
...
+            prim->clut = entity->palette + 1;
+
+            ptr = &D_80183028[params * 4];
+            prim->u0 = prim->u2 = *ptr++ + ((primIndex & 1) << 7);
+            prim->u1 = prim->u3 = *ptr++ + ((primIndex & 1) << 7);
+            prim->v0 = prim->v1 = *ptr++ + ((primIndex & 2) << 6);
+            prim->v2 = prim->v3 = *ptr + ((primIndex & 2) << 6);
+            prim->g0 = prim->g1 = prim->g2 = prim->g3 = prim->r0 = prim->r1 =
+                prim->r2 = prim->r3 = 0x80;
+            prim->b0 = prim->b2 = prim->b1 = prim->b3 = 0x80;
```
Comment (JoshSchreuder) [inline src/st/no3/4AFF0.c:359]: Whoops, I'll inline this as well

Comment (JoshSchreuder) [reply src/st/no3/4AFF0.c:359]: Actually do we prefer these inlined or separated out?

Inlined would be
```
            prim->b0 = prim->b2 = prim->b1 = prim->b3 = prim->g0 = prim->g1 =
                prim->g2 = prim->g3 = prim->r0 = prim->r1 = prim->r2 =
                    prim->r3 = 0x80;
```

With the column breaks

vs

```
            prim->r0 = prim->r1 = prim->r2 = prim->r3 = 0x80;
            prim->g0 = prim->g1 = prim->g2 = prim->g3 = 0x80;
            prim->b0 = prim->b2 = prim->b1 = prim->b3 = 0x80;
```

File: src/st/no3/4AFF0.c [hunk 2]
```diff
@@ -319,7 +319,127 @@ void EntityUnkId31(Entity* self) {
...
+            entity->velocityX = *ptr32++;
+        }
+        entity->velocityY = *ptr32;
+    }
+
+    posX = entity->posX.i.hi;
+    posY = entity->posY.i.hi;
+    point = &D_80183054[params];
+    prim = &g_PrimBuf[entity->primIndex];
+    entity->ext.factory.unk7E += D_8018304C[params];
```
Comment (sozud) [inline src/st/no3/4AFF0.c:382]: We're mixing ext.timer and ext.factory, we should probably make an extension

File: src/st/no3/4AFF0.c (hunk 1 above)
Comment (sozud) [reply src/st/no3/4AFF0.c:359]: I don't think those are equivalent on psp?

Comment (JoshSchreuder) [reply src/st/no3/4AFF0.c:359]: Yep you're right, looks like PSP is more happy with the latter, so I'll update to that

File: src/st/no3/4AFF0.c (hunk 2 above)
Comment (JoshSchreuder) [reply src/st/no3/4AFF0.c]: Added, hope this is what you meant

File: src/st/no3/4AFF0.c (hunk 1 above)
Comment (JoshSchreuder) [reply src/st/no3/4AFF0.c:359]: Updated, and also fixed up a few other things like `if (self->step != 0)` to be a closer match on the PSP side

**Pull request merged**